src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Synthesized.cs

```diff
@@ -1653,5 +1653,116 @@ class B : A
                 AssertEx.SetEqual(new[] { "CompilerGeneratedAttribute", "DebuggerHiddenAttribute" }, GetAttributeNames(baseMethodWrapper.GetAttributes()));
             }
         }
+
+        [Fact, WorkItem(7809, "https://github.com/dotnet/roslyn/issues/7809")]
+        public void SynthesizeAttributeWithUseSiteErrorFails()
+        {
+            #region "mslib"
+            var mslibNoString = @"
+namespace System
+{
+    public class Object {}
+    public struct Byte { }
+    public struct Int16 { }
+    public struct Int32 { }
+    public struct Int64 { }
+    public struct Single { }
+    public struct Double { }
+    public struct SByte { }
+    public struct UInt32 { }
+    public struct UInt64 { }
+    public struct Char { }
+    public struct Boolean { }
+    public struct UInt16 { }
+    public struct UIntPtr { }
+    public struct IntPtr { }
+    public class Delegate { }
+    public class MulticastDelegate { }
+    public class Array { }
+    public class Exception { }
+    public class Type { }
+    public class ValueType { }
+    public class Enum { }
+    public class Attribute { }
+    public class ParamArrayAttribute { }
+    public struct Void { }
+    public struct RuntimeFieldHandle { }
+    public struct RuntimeTypeHandle { }
+    public class Activator
+    {
+         public static T CreateInstance<T>() { return default(T); }
+    }
+    public class NotSupportedException : Exception {}
+
+    namespace Runtime
+    {
+        namespace InteropServices
+        {
+            public class OutAttribute { }
+        }
+    }
+
+    namespace Reflection
+    {
+        public class DefaultMemberAttribute { }
+    }
+}
+";
+            var mslib = mslibNoString + @"
+namespace System
+{
+    public class String
+    {
+        public int Length { get { return 10; } }
+    }
+}";
+            #endregion
+
+            // Build an mscorlib including String
+            var mslibComp = CreateCompilation(new string[] { mslib });
+            var diag = mslibComp.GetDiagnostics();
+            Assert.False(diag.Any(d => d.Severity == DiagnosticSeverity.Error));
```

Comment:
Consider using `mslibComp.VerifyDiagnostics()` instead. Similar comment for other compilations below.